src/idl_parser.cpp

```diff
@@ -2004,18 +2048,39 @@ CheckedError Parser::DoParse(const char *source, const char **include_paths,
   while (token_ != kTokenEof) {
     if (opts.proto_mode) {
       ECHECK(ParseProtoDecl());
-    } else if (token_ == kTokenNameSpace) {
-      ECHECK(ParseNamespace());
-    } else if (token_ == '{') {
+  }
+  else if (token_ == kTokenNameSpace) {
+    ECHECK(ParseNamespace());
+
+  } else if (token_ == '{' || token_ == '[') {
       if (!root_struct_def_)
         return Error("no root type set to parse json with");
       if (builder_.GetSize()) {
         return Error("cannot have more than one json object in a file");
       }
-      uoffset_t toff;
+
+    uoffset_t toff;
+
+    const bool top_level_list = token_ == '[';
+    if (top_level_list)
+    {
+      if(root_struct_def_->fields.vec.size() > 1)
+        return Error("top level list found with more than one field in schema");
```

Comment:
In the top level list case (illustrated in my comment just above), json effectively omits the name of the vector field from the root type object we are trying to map it to. 
That checks makes sure there is no ambiguity introduced by skipping the field name: it enforces there is a single field in the root type object.